Pipfile

```diff
@@ -4,22 +4,23 @@ verify_ssl = true
 name = "pypi"
 
 [scripts]
-test = "nosetests"
 lint = "flake8 ."
+publish = "python setup.py sdist upload"
+test = "nosetests"
 
 [dev-packages]
-"flake8-debugger" = ">=3.1"
-"flake8-quotes" = ">=0.14.0"
-"flake8" = ">=3.5"
-"pep8-naming" = "==0.5.0"
-coverage = ">=4.4"
-nose = ">=1.3"
-rednose = ">=1.2"
-sure = ">=1.4"
+"flake8-debugger" = "*"
+"flake8-quotes" = "*"
+"flake8" = "*"
+"pep8-naming" = "*"
+coverage = "*"
+nose = "*"
+rednose = "*"
+sure = "*"
```

Comment:
why are this `*` instead of a version. I get that `*` means any version but, why is that what we want?

Reply:
The reasoning behind this is that:
* This is a library. The application is the one that should pin the dependency according to its requirements.
* For development, the lock file keeps the dependencies pinned.

Reply:
Basically, you don't want to pin your dependencies until you find something that breaks. In python, semantic versioning is not cannon and there's no actual way of knowing when a future upgrade will or won't break your changes. It doesn't make sens to pin your dependencies until you run into such an issue.

Even with semantic versioning, if something is backwards incompatible, it doesn't mean that what you use is incompatible. Your code could very easily work with version 1.x or 2.x without changes, so you don't want to limit your code that way.